### PR TITLE
Change RLM parsing to use product.feature instead of product_feature

### PR DESCRIPTION
**Pull request by julianaklulo**
#### What
Change the designation of product and feature to use `.` instead of `_`as separator.

#### Why
To keep the same nomenclature across different license servers, such as flexlm.

`Task`: https://app.clickup.com/t/18022949/LM-148

### CODE REVIEW

File: agent/tests/test_tokenstat.py [hunk 1]
```diff
@@ -248,27 +248,24 @@ async def test_report_rlm(
     reconcile_list = await tokenstat.report()
     assert reconcile_list == [
```
Comment (zsinx6) [inline agent/tests/test_tokenstat.py]: What do you think about creating more tests for this? With an empty config table and empty used_licenses

Comment (julianaklulo) [reply agent/tests/test_tokenstat.py]: I refactored the test to use parametrize, this way we can test both the case with checked out licenses and the empty one

Comment (julianaklulo) [reply agent/tests/test_tokenstat.py]: Also added another test for empty backend config.

**Review by zsinx6 — APPROVED** [review]
